### PR TITLE
Add fastDistAA and test for it

### PR DESCRIPTION
**Pull request by Vivian0105**

### CODE REVIEW

File: tests/testthat/testfastDistAA.R [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#### fastDistAA vs pairwiseDist ####
+
+test_that("fastDistAA matches pairwiseDist for amimo acid sequences", {
```
Comment (ggabernet) [inline tests/testthat/testfastDistAA.R]:
```suggestion
test_that("fastDistAA matches pairwiseDist for amino acid sequences", {
```

File: tests/testthat/testfastDistAA.R [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            "*-.XRSTVWY",
+            "*CDERSTVWY")
+  
+  fast_distAA <- alakazam:::fastDistAA(seqs)
+  fast_distAA <- as.matrix(fast_distAA) 
+  pw_dist<- alakazam::pairwiseDist(seq = seqs, dist_mat = aa_mat)
+  
+  # Same results when comparing to pairwiseDist with check.attributes=F (ignoring dimnames)
+  expect_equal(fast_distAA, pw_dist, check.attributes=F)
+
```
Comment (ggabernet) [inline tests/testthat/testfastDistAA.R]: Could you also add tests that check what happens if only one sequence is in the original matrix (empty distance matrix should be returned) and what happens if the original list is empty (throwing clean error that the sequence list is empty).

File: R/Sequence.R [hunk 3]
```diff
@@ -1204,6 +1204,32 @@ fastDist <- function(seqs) {
 }
 
 
+#' Faster calculation of pairwise distances between amino acid sequences of the same length
+#' 
+#' \code{fastDistAA} calculates all pairwise distance between a set of amino acid sequences of the same length.
```
Comment (ggabernet) [inline R/Sequence.R]:
```suggestion
#' \code{fastDistAA} calculates all pairwise distances between a set of amino acid sequences of the same length. 
```

Comment (ggabernet) [reply R/Sequence.R]: Could the documentation here also specify the matching rules for special characters (X, ., - , *) and that IUPAC characters are not supported (only 20 canonical amino acids).

**Review by ggabernet — COMMENTED** [review]
I just have some comments, otherwise looks good to me!

**Review by ggabernet — APPROVED** [review]
LGTM now!
